### DISCUSSION TITLE
Simulation for System Evaluation

### DISCUSSION THREAD

**Discussion by matt-ashman** [Sky360-Repository/sky360#6] [Ideas]
Has anyone attempted to use simulated synthetic data to evaluate the existing system and compare it with alternatives?

Imagine using a software package like Unity or Omniverse to create simulated sensor inputs and running those through at least the tracking component of the software stack. You could create simulated targets (birds, drones, aircraft, UAP) recorded through simulated cameras, testing the limits of the tracker under various frame rates with "realistic" motions.

**Comment by i-am-logger** [comment]
@Micky-G will be the best to answer this.

**Comment by Micky-G** [comment]
Hey @matt-ashman, the short answer is no, we have not got that far yet. This is certainly something we can add to the road map as it does sound like it would help us greatly. When you have a couple of minutes, would you be able to include some sort of article or blog post on how we would go about achieving this please?

**Reply by matt-ashman** [reply]
Yes, absolutely. I'll start by summarizing the general approach and then link into related materials.

First of all, you need a simulation platform. I have used Unity 3D to great success for game development in the past, and it might be most helpful to think of this simulation as a game-like scenario.

https://unity.com/

There are alternatives like Omniverse that are incredibly capable, but with a bit more complexity to manage for simple projects.

https://www.nvidia.com/en-us/omniverse/?nvid=nv-int-solr-460533-vt08#cid=_nv-int-solr_en-us

Think of the player of the game as being the detection/tracking software system with a simulated sensor (i.e. camera projection). The player "wins" when it successfully detect and tracks objects, and it "loses" when it fails to detect or loses track. The aggregate score can be computed over a large number of simulated scenarios as a metric for how well the system performs.

The basic elements to define for the simulation are:
1. Data contract between simulation and tracker software (timestamps, images, etc.)
2. Methods for sensor simulation, i.e. synthetic camera image formation (projection)
3. Simulated environment, what does your "world" look like and where are the sensors?
4. Simulated actors (i.e. airplanes, bugs, birds, UAP) and trajectory generation

Since you can run the tracker offline with video, you could start by just recording synthetic videos for evaluation purposes. There isn't a real need to run the tracker "in the loop" until you start to simulate other elements like the PTZ camera with physical dynamics.

**Reply by Micky-G** [reply]
Thank you very much. Will have a read and try to better my understanding on this sort of thing.

**Reply by matt-ashman** [reply]
Maybe I can help get things started with a simple simulation demo to demonstrate these elements?

**Reply by Micky-G** [reply]
That would be greatly appreciated, thank you very much.

**Comment by matt-ashman** [comment]
Assuming Unity for simplicity, check this out:

https://learn.unity.com/

Actually, the most complex thing about this might be simulating the fisheye camera projection accurately.
Otherwise I think someone could get a basic simulation up and running very quickly after taking this tutorial.

**Reply by Micky-G** [reply]
Thanks!
